### DISCUSSION TITLE
What format does png info need to be in order to be read by civitai.

### DISCUSSION THREAD

**Discussion by xzuyn** [civitai/civitai#249] [Q&A]
Currently trying to fix onnxdiffusersui to use the proper formatting for images to be auto filled into civitai. 
At the moment only seed and guidance scale get read.

example image info/current formatting:
`((masterpiece)), (best quality), (detailed), (1girl), looking at viewer, (((glasses))), hearts, long_hair, blush, crossed_arms Negative prompt: (worst quality, low quality:1.4) Steps: 16, Sampler: DEISMultistepScheduler, CFG scale: 3.5, Seed: 1636388198, Model: AniMerge_v1-0_MaxSlicing_CLIPSKIP_onnx-fp16, Iteration size: 1, batch size: 1, eta: 0,`

what civitai fills into the site:

`seed: 1636388198
guidance scale: 3.5`

**Comment by JustMaier** [comment]
Hey, we read things in the Automatic1111 format. You can [see the logic here](https://github.com/civitai/civitai/blob/7b042c50170bc504a82a84b4f171296d774e1612/src/utils/image-metadata.ts#L71).

Basically it's:
```
{Prompt Text}
Negative prompt: {Negative Prompt}
{Other Key: Value pairs separated by a comma}
```

Looking at what you have, I think the only thing you'd need to adjust is moving the Steps and other KV pairs to the next line like so:
```
((masterpiece)), (best quality), (detailed), (1girl), looking at viewer, (((glasses))), hearts, long_hair, blush, crossed_arms
Negative prompt: (worst quality, low quality:1.4)
Steps: 16, Sampler: DEISMultistepScheduler, CFG scale: 3.5, Seed: 1636388198, Model: AniMerge_v1-0_MaxSlicing_CLIPSKIP_onnx-fp16, Iteration size: 1, batch size: 1, eta: 0
```

**Reply by xzuyn** [reply]
thank you. i will try this out later.

**Reply by xzuyn** [reply]
i cant seem to figure it out. looking at other users info its all one line like; `Parameters: A middle-aged elven woman with a kind expression, waist-length silver hair styled in loose waves, soft hazel eyes, and a gentle smile, sitting on a bench in a quiet park, surrounded by the sounds of the city, lost in thought and soaking up the peace and tranquility., (((full detailed background))).Negative prompt: 3d, cg, extra hands, extra fingers, extra limbs, censored, text, watermark,hat, black and white, ((piercings)), ((piercing)), ((tatoo)), ((pubic hair)), (((blind))), (worst quality, low quality:1.4), extra hands, extra limbs, extra fingers, long neck, .Steps: 24, Sampler: DPM++ SDE Karras, CFG scale: 3.5, Seed: 2434918121, Size: 704x1024, Model hash: d9caabae00, Model: animerge_v10, Clip skip: 2` which loads fine.

i've tried even having each parameter be its own section like `metadata.add_text("Negative prompt: ", str(neg_prompt))`, which looks much nicer and is easier for a human to read through exiftool, but civitai doesnt read it at all.

**Reply by xzuyn** [reply]
i feel as though this formatting with each parameter on its own section should be supported as a big block "parameters" in one section is not visually appealing and hard to read.

`Prompt: a photo of a dog on a beach`
`Negative Prompt: (worst quality, low quality:1.4), watermark, signature, logo, text, lowres, teeth, ((monochrome, grayscale))`
`Steps: 16`
`Sampler: DEISMultistepScheduler`
`CFG Scale: 3.5`
`Seed: 2271430027`
`Size: 512x768`
`Model: PhotoMerge_v1-2_MaxSlicing_Optimized_ft_mse_onnx-fp16`
`Iteration Size: 1`
`Batch Size: 1`
`Eta: 0`
![000045-00 a photo of a dog on a beach_2271430027_3 5g_512x768_16s_DEIS](https://user-images.githubusercontent.com/16216325/218689309-4ee95205-a9af-4b62-a083-9611ae2d5471.png)
